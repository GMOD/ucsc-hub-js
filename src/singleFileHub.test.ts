import fs from 'fs'
import { test, expect } from 'vitest'
import SingleFileHub from './singleFileHub.ts'

// test that singleFileHub can parse singleFileAssemblyHub.txt
test('single-file hub', () => {
  // NOTE: this file was slightly modified from the
  // original to fix inconsistent indentation
  const input = fs.readFileSync(
    'test/singleFileAssemblyHub.GCF_000002985.6.hub.txt',
    'utf8',
  )
  const singleFileHub = new SingleFileHub(input)
  expect(singleFileHub).toMatchSnapshot()
  expect(singleFileHub.genome).toMatchSnapshot()
  expect(singleFileHub.tracks).toMatchSnapshot()
})
