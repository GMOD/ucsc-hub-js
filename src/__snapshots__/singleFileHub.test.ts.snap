// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`single-file hub 1`] = `
SingleFileHub {
  "genome": RaStanza {
    "data": {
      "blat": "dynablat-01.soe.ucsc.edu 4040 dynamic GCF/000/002/985/GCF_000002985.6",
      "chromAliasBb": "GCF_000002985.6.chromAlias.bb",
      "chromAuthority": "ucsc",
      "chromSizes": "GCF_000002985.6.chrom.sizes.txt",
      "defaultPos": "NC_003283.11:6974726-6984726",
      "description": "roundworm",
      "genome": "GCF_000002985.6",
      "groups": "groups.txt",
      "htmlPath": "html/GCF_000002985.6_WBcel235.description.html",
      "isPcr": "dynablat-01.soe.ucsc.edu 4040 dynamic GCF/000/002/985/GCF_000002985.6",
      "organism": "WBcel235 Feb. 2013",
      "scientificName": "Caenorhabditis elegans",
      "taxId": "6239",
      "transBlat": "dynablat-01.soe.ucsc.edu 4040 dynamic GCF/000/002/985/GCF_000002985.6",
      "twoBitBptUrl": "GCF_000002985.6.2bit.bpt",
      "twoBitPath": "GCF_000002985.6.2bit",
    },
    "name": "GCF_000002985.6",
    "nameKey": "genome",
  },
  "hubData": RaStanza {
    "data": {
      "descriptionUrl": "html/GCF_000002985.6_WBcel235.description.html",
      "email": "hclawson@ucsc.edu",
      "hub": "GCF_000002985.6 genome assembly",
      "longLabel": "roundworm/Caenorhabditis elegans/GCF_000002985.6_WBcel235 genome assembly",
      "shortLabel": "roundworm",
      "useOneFile": "on",
    },
    "name": "GCF_000002985.6 genome assembly",
    "nameKey": "hub",
  },
  "tracks": TrackDbFile {
    "data": {
      "assembly": RaStanza {
        "data": {
          "altColor": "230,170,40",
          "bigDataUrl": "bbi/GCF_000002985.6_WBcel235.assembly.bb",
          "color": "150,100,30",
          "colorByStrand": "150,100,30 230,170,40",
          "group": "map",
          "html": "html/GCF_000002985.6_WBcel235.assembly",
          "longLabel": "Assembly",
          "searchIndex": "name",
          "searchTrix": "ixIxx/GCF_000002985.6_WBcel235.assembly.ix",
          "shortLabel": "Assembly",
          "track": "assembly",
          "type": "bigBed 6",
          "url": "https://www.ncbi.nlm.nih.gov/nuccore/$$",
          "urlLabel": "NCBI Nucleotide database:",
          "visibility": "pack",
        },
        "name": "assembly",
        "nameKey": "track",
      },
      "augustus": RaStanza {
        "data": {
          "bigDataUrl": "bbi/GCF_000002985.6_WBcel235.augustus.bb",
          "color": "180,0,0",
          "group": "genes",
          "html": "html/GCF_000002985.6_WBcel235.augustus",
          "longLabel": "Augustus Gene Predictions",
          "shortLabel": "Augustus",
          "track": "augustus",
          "type": "bigGenePred",
          "visibility": "dense",
        },
        "name": "augustus",
        "nameKey": "track",
      },
      "cpgIslandExt": RaStanza {
        "data": {
          "bigDataUrl": "bbi/GCF_000002985.6_WBcel235.cpgIslandExt.bb",
          "longLabel": "CpG Islands (Islands < 300 Bases are Light Green)",
          "parent": "cpgIslands off",
          "priority": "1",
          "shortLabel": "CpG Islands",
          "track": "cpgIslandExt",
          "type": "bigBed 4 +",
        },
        "name": "cpgIslandExt",
        "nameKey": "track",
      },
      "cpgIslandExtUnmasked": RaStanza {
        "data": {
          "bigDataUrl": "bbi/GCF_000002985.6_WBcel235.cpgIslandExtUnmasked.bb",
          "longLabel": "CpG Islands on All Sequence (Islands < 300 Bases are Light Green)",
          "parent": "cpgIslands on",
          "priority": "2",
          "shortLabel": "Unmasked CpG",
          "track": "cpgIslandExtUnmasked",
          "type": "bigBed 4 +",
        },
        "name": "cpgIslandExtUnmasked",
        "nameKey": "track",
      },
      "cpgIslands": RaStanza {
        "data": {
          "compositeTrack": "on",
          "group": "regulation",
          "html": "html/GCF_000002985.6_WBcel235.cpgIslands",
          "longLabel": "CpG Islands (Islands < 300 Bases are Light Green)",
          "shortLabel": "CpG Islands",
          "track": "cpgIslands",
          "type": "bigBed 4 +",
          "visibility": "pack",
        },
        "name": "cpgIslands",
        "nameKey": "track",
      },
      "cytoBandIdeo": RaStanza {
        "data": {
          "bigDataUrl": "bbi/GCF_000002985.6_WBcel235.cytoBand.bb",
          "group": "map",
          "longLabel": "Ideogram for Orientation",
          "shortLabel": "Chromosome Band (Ideogram)",
          "track": "cytoBandIdeo",
          "type": "bigBed 4 +",
          "visibility": "dense",
        },
        "name": "cytoBandIdeo",
        "nameKey": "track",
      },
      "ensGene": RaStanza {
        "data": {
          "bigDataUrl": "bbi/GCF_000002985.6_WBcel235.ensGene.bb",
          "color": "150,0,0",
          "group": "genes",
          "html": "html/GCF_000002985.6_WBcel235.ensGene",
          "longLabel": "Ensembl genes version 103/Feb. 2021",
          "priority": "40",
          "searchIndex": "name",
          "searchTrix": "ixIxx/GCF_000002985.6_WBcel235.ensGene.ix",
          "shortLabel": "Ensembl genes",
          "track": "ensGene",
          "type": "bigBed 12 .",
          "visibility": "pack",
        },
        "name": "ensGene",
        "nameKey": "track",
      },
      "gc5Base": RaStanza {
        "data": {
          "altColor": "128,128,128",
          "autoScale": "Off",
          "bigDataUrl": "bbi/GCF_000002985.6_WBcel235.gc5Base.bw",
          "color": "0,0,0",
          "graphTypeDefault": "Bar",
          "gridDefault": "OFF",
          "group": "map",
          "html": "html/GCF_000002985.6_WBcel235.gc5Base",
          "longLabel": "GC Percent in 5-Base Windows",
          "maxHeightPixels": "128:36:16",
          "shortLabel": "GC Percent",
          "track": "gc5Base",
          "type": "bigWig 0 100",
          "viewLimits": "30:70",
          "visibility": "full",
          "windowingFunction": "Mean",
        },
        "name": "gc5Base",
        "nameKey": "track",
      },
      "ncbiRefSeq": RaStanza {
        "data": {
          "altColor": "120,12,12",
          "baseColorDefault": "genomicCodons",
          "baseColorUseCds": "given",
          "bigDataUrl": "bbi/GCF_000002985.6_WBcel235.ncbiRefSeq.bb",
          "color": "12,12,120",
          "defaultLabelFields": "geneName2",
          "idXref": "ncbiRefSeqLink mrnaAcc name",
          "labelFields": "name,geneName,geneName2",
          "longLabel": "NCBI RefSeq genes, curated and predicted sets (NM_*, XM_*, NR_*, XR_*, NP_* or YP_*)",
          "parent": "refSeqComposite on",
          "priority": "1",
          "searchIndex": "name",
          "searchTrix": "ixIxx/GCF_000002985.6_WBcel235.ncbiRefSeq.ix",
          "shortLabel": "RefSeq All",
          "track": "ncbiRefSeq",
          "type": "bigGenePred",
        },
        "name": "ncbiRefSeq",
        "nameKey": "track",
      },
      "ncbiRefSeqCurated": RaStanza {
        "data": {
          "baseColorDefault": "genomicCodons",
          "baseColorUseCds": "given",
          "bigDataUrl": "bbi/GCF_000002985.6_WBcel235.ncbiRefSeqCurated.bb",
          "color": "12,12,120",
          "defaultLabelFields": "geneName2",
          "idXref": "ncbiRefSeqLink mrnaAcc name",
          "labelFields": "name,geneName,geneName2",
          "longLabel": "NCBI RefSeq genes, curated subset (NM_*, NR_*, NP_* or YP_*)",
          "parent": "refSeqComposite on",
          "priority": "2",
          "searchIndex": "name",
          "searchTrix": "ixIxx/GCF_000002985.6_WBcel235.ncbiRefSeqCurated.ix",
          "shortLabel": "RefSeq Curated",
          "track": "ncbiRefSeqCurated",
          "type": "bigGenePred",
        },
        "name": "ncbiRefSeqCurated",
        "nameKey": "track",
      },
      "ncbiRefSeqOther": RaStanza {
        "data": {
          "bigDataUrl": "bbi/GCF_000002985.6_WBcel235.ncbiRefSeqOther.bb",
          "color": "32,32,32",
          "labelFields": "name",
          "longLabel": "NCBI RefSeq other annotations (not NM_*, NR_*, XM_*, XR_*, NP_* or YP_*)",
          "parent": "refSeqComposite on",
          "priority": "4",
          "searchIndex": "name",
          "searchTrix": "ixIxx/GCF_000002985.6_WBcel235.ncbiRefSeqOther.ix",
          "shortLabel": "RefSeq Other",
          "skipEmptyFields": "on",
          "track": "ncbiRefSeqOther",
          "type": "bigBed 12 +",
          "urls": "GeneID="https://www.ncbi.nlm.nih.gov/gene/$$" MIM="https://www.ncbi.nlm.nih.gov/omim/$$" HGNC="http://www.genenames.org/cgi-bin/gene_symbol_report?hgnc_id=$$" FlyBase="http://flybase.org/reports/$$" WormBase="http://www.wormbase.org/db/gene/gene?name=$$" RGD="https://rgd.mcw.edu/rgdweb/search/search.html?term=$$" SGD="https://www.yeastgenome.org/locus/$$" miRBase="http://www.mirbase.org/cgi-bin/mirna_entry.pl?acc=$$" ZFIN="https://zfin.org/$$" MGI="http://www.informatics.jax.org/marker/$$"",
        },
        "name": "ncbiRefSeqOther",
        "nameKey": "track",
      },
      "ncbiRefSeqPsl": RaStanza {
        "data": {
          "baseColorDefault": "diffCodons",
          "baseColorUseCds": "table given",
          "baseColorUseSequence": "lfExtra",
          "bigDataUrl": "bbi/GCF_000002985.6_WBcel235.bigPsl.bb",
          "color": "0,0,0",
          "indelDoubleInsert": "on",
          "indelQueryInsert": "on",
          "longLabel": "RefSeq Alignments of RNAs",
          "parent": "refSeqComposite off",
          "priority": "5",
          "pslSequence": "no",
          "searchIndex": "name",
          "shortLabel": "RefSeq Alignments",
          "showCdsAllScales": ".",
          "showCdsMaxZoom": "10000.0",
          "showDiffBasesAllScales": ".",
          "showDiffBasesMaxZoom": "10000.0",
          "track": "ncbiRefSeqPsl",
          "type": "bigPsl",
        },
        "name": "ncbiRefSeqPsl",
        "nameKey": "track",
      },
      "refSeqComposite": RaStanza {
        "data": {
          "allButtonPair": "on",
          "compositeTrack": "on",
          "dataVersion": "WormBase WS277 (2020-05-29)",
          "dragAndDrop": "subTracks",
          "group": "genes",
          "html": "html/GCF_000002985.6_WBcel235.refSeqComposite",
          "longLabel": "RefSeq gene predictions from NCBI",
          "priority": "2",
          "shortLabel": "NCBI RefSeq",
          "track": "refSeqComposite",
          "type": "bigBed",
          "visibility": "pack",
        },
        "name": "refSeqComposite",
        "nameKey": "track",
      },
      "repeatMasker": RaStanza {
        "data": {
          "bigDataUrl": "bbi/GCF_000002985.6_WBcel235.rmsk.bb",
          "group": "varRep",
          "html": "html/GCF_000002985.6_WBcel235.repeatMasker",
          "longLabel": "RepeatMasker Repetitive Elements",
          "maxWindowToDraw": "5000000",
          "shortLabel": "RepeatMasker",
          "track": "repeatMasker",
          "type": "bigRmsk 9 +",
          "visibility": "pack",
        },
        "name": "repeatMasker",
        "nameKey": "track",
      },
      "simpleRepeat": RaStanza {
        "data": {
          "bigDataUrl": "bbi/GCF_000002985.6_WBcel235.simpleRepeat.bb",
          "group": "varRep",
          "html": "html/GCF_000002985.6_WBcel235.simpleRepeat",
          "longLabel": "Simple Tandem Repeats by TRF",
          "shortLabel": "Simple Repeats",
          "track": "simpleRepeat",
          "type": "bigBed 4 +",
          "visibility": "dense",
        },
        "name": "simpleRepeat",
        "nameKey": "track",
      },
      "tanDups": RaStanza {
        "data": {
          "compositeTrack": "on",
          "group": "map",
          "html": "html/GCF_000002985.6_WBcel235.tanDups",
          "longLabel": "Paired identical sequences",
          "shortLabel": "Tandem Dups",
          "track": "tanDups",
          "type": "bigBed 12",
          "visibility": "hide",
        },
        "name": "tanDups",
        "nameKey": "track",
      },
      "tandemDups": RaStanza {
        "data": {
          "bigDataUrl": "bbi/GCF_000002985.6_WBcel235.tandemDups.bb",
          "longLabel": "Paired exactly identical sequence survey over entire genome assembly",
          "parent": "tanDups on",
          "shortLabel": "Tandem Dups",
          "track": "tandemDups",
          "type": "bigBed 12",
        },
        "name": "tandemDups",
        "nameKey": "track",
      },
      "windowMasker": RaStanza {
        "data": {
          "bigDataUrl": "bbi/GCF_000002985.6_WBcel235.windowMasker.bb",
          "group": "varRep",
          "html": "html/GCF_000002985.6_WBcel235.windowMasker",
          "longLabel": "Genomic Intervals Masked by WindowMasker + SDust",
          "shortLabel": "WM + SDust",
          "track": "windowMasker",
          "type": "bigBed 3",
          "visibility": "dense",
        },
        "name": "windowMasker",
        "nameKey": "track",
      },
      "xenoRefGene": RaStanza {
        "data": {
          "bigDataUrl": "bbi/GCF_000002985.6_WBcel235.xenoRefGene.bb",
          "color": "180,0,0",
          "defaultLabelFields": "geneName",
          "group": "rna",
          "html": "html/GCF_000002985.6_WBcel235.xenoRefGene",
          "labelFields": "name,geneName,geneName2",
          "longLabel": "RefSeq mRNAs mapped to this assembly",
          "searchIndex": "name",
          "searchTrix": "ixIxx/GCF_000002985.6_WBcel235.xenoRefGene.ix",
          "shortLabel": "RefSeq mRNAs",
          "track": "xenoRefGene",
          "type": "bigGenePred",
          "url": "https://www.ncbi.nlm.nih.gov/nuccore/$$",
          "urlLabel": "NCBI Nucleotide database:",
          "visibility": "pack",
        },
        "name": "xenoRefGene",
        "nameKey": "track",
      },
    },
    "nameKey": "track",
  },
}
`;

exports[`single-file hub 2`] = `
RaStanza {
  "data": {
    "blat": "dynablat-01.soe.ucsc.edu 4040 dynamic GCF/000/002/985/GCF_000002985.6",
    "chromAliasBb": "GCF_000002985.6.chromAlias.bb",
    "chromAuthority": "ucsc",
    "chromSizes": "GCF_000002985.6.chrom.sizes.txt",
    "defaultPos": "NC_003283.11:6974726-6984726",
    "description": "roundworm",
    "genome": "GCF_000002985.6",
    "groups": "groups.txt",
    "htmlPath": "html/GCF_000002985.6_WBcel235.description.html",
    "isPcr": "dynablat-01.soe.ucsc.edu 4040 dynamic GCF/000/002/985/GCF_000002985.6",
    "organism": "WBcel235 Feb. 2013",
    "scientificName": "Caenorhabditis elegans",
    "taxId": "6239",
    "transBlat": "dynablat-01.soe.ucsc.edu 4040 dynamic GCF/000/002/985/GCF_000002985.6",
    "twoBitBptUrl": "GCF_000002985.6.2bit.bpt",
    "twoBitPath": "GCF_000002985.6.2bit",
  },
  "name": "GCF_000002985.6",
  "nameKey": "genome",
}
`;

exports[`single-file hub 3`] = `
TrackDbFile {
  "data": {
    "assembly": RaStanza {
      "data": {
        "altColor": "230,170,40",
        "bigDataUrl": "bbi/GCF_000002985.6_WBcel235.assembly.bb",
        "color": "150,100,30",
        "colorByStrand": "150,100,30 230,170,40",
        "group": "map",
        "html": "html/GCF_000002985.6_WBcel235.assembly",
        "longLabel": "Assembly",
        "searchIndex": "name",
        "searchTrix": "ixIxx/GCF_000002985.6_WBcel235.assembly.ix",
        "shortLabel": "Assembly",
        "track": "assembly",
        "type": "bigBed 6",
        "url": "https://www.ncbi.nlm.nih.gov/nuccore/$$",
        "urlLabel": "NCBI Nucleotide database:",
        "visibility": "pack",
      },
      "name": "assembly",
      "nameKey": "track",
    },
    "augustus": RaStanza {
      "data": {
        "bigDataUrl": "bbi/GCF_000002985.6_WBcel235.augustus.bb",
        "color": "180,0,0",
        "group": "genes",
        "html": "html/GCF_000002985.6_WBcel235.augustus",
        "longLabel": "Augustus Gene Predictions",
        "shortLabel": "Augustus",
        "track": "augustus",
        "type": "bigGenePred",
        "visibility": "dense",
      },
      "name": "augustus",
      "nameKey": "track",
    },
    "cpgIslandExt": RaStanza {
      "data": {
        "bigDataUrl": "bbi/GCF_000002985.6_WBcel235.cpgIslandExt.bb",
        "longLabel": "CpG Islands (Islands < 300 Bases are Light Green)",
        "parent": "cpgIslands off",
        "priority": "1",
        "shortLabel": "CpG Islands",
        "track": "cpgIslandExt",
        "type": "bigBed 4 +",
      },
      "name": "cpgIslandExt",
      "nameKey": "track",
    },
    "cpgIslandExtUnmasked": RaStanza {
      "data": {
        "bigDataUrl": "bbi/GCF_000002985.6_WBcel235.cpgIslandExtUnmasked.bb",
        "longLabel": "CpG Islands on All Sequence (Islands < 300 Bases are Light Green)",
        "parent": "cpgIslands on",
        "priority": "2",
        "shortLabel": "Unmasked CpG",
        "track": "cpgIslandExtUnmasked",
        "type": "bigBed 4 +",
      },
      "name": "cpgIslandExtUnmasked",
      "nameKey": "track",
    },
    "cpgIslands": RaStanza {
      "data": {
        "compositeTrack": "on",
        "group": "regulation",
        "html": "html/GCF_000002985.6_WBcel235.cpgIslands",
        "longLabel": "CpG Islands (Islands < 300 Bases are Light Green)",
        "shortLabel": "CpG Islands",
        "track": "cpgIslands",
        "type": "bigBed 4 +",
        "visibility": "pack",
      },
      "name": "cpgIslands",
      "nameKey": "track",
    },
    "cytoBandIdeo": RaStanza {
      "data": {
        "bigDataUrl": "bbi/GCF_000002985.6_WBcel235.cytoBand.bb",
        "group": "map",
        "longLabel": "Ideogram for Orientation",
        "shortLabel": "Chromosome Band (Ideogram)",
        "track": "cytoBandIdeo",
        "type": "bigBed 4 +",
        "visibility": "dense",
      },
      "name": "cytoBandIdeo",
      "nameKey": "track",
    },
    "ensGene": RaStanza {
      "data": {
        "bigDataUrl": "bbi/GCF_000002985.6_WBcel235.ensGene.bb",
        "color": "150,0,0",
        "group": "genes",
        "html": "html/GCF_000002985.6_WBcel235.ensGene",
        "longLabel": "Ensembl genes version 103/Feb. 2021",
        "priority": "40",
        "searchIndex": "name",
        "searchTrix": "ixIxx/GCF_000002985.6_WBcel235.ensGene.ix",
        "shortLabel": "Ensembl genes",
        "track": "ensGene",
        "type": "bigBed 12 .",
        "visibility": "pack",
      },
      "name": "ensGene",
      "nameKey": "track",
    },
    "gc5Base": RaStanza {
      "data": {
        "altColor": "128,128,128",
        "autoScale": "Off",
        "bigDataUrl": "bbi/GCF_000002985.6_WBcel235.gc5Base.bw",
        "color": "0,0,0",
        "graphTypeDefault": "Bar",
        "gridDefault": "OFF",
        "group": "map",
        "html": "html/GCF_000002985.6_WBcel235.gc5Base",
        "longLabel": "GC Percent in 5-Base Windows",
        "maxHeightPixels": "128:36:16",
        "shortLabel": "GC Percent",
        "track": "gc5Base",
        "type": "bigWig 0 100",
        "viewLimits": "30:70",
        "visibility": "full",
        "windowingFunction": "Mean",
      },
      "name": "gc5Base",
      "nameKey": "track",
    },
    "ncbiRefSeq": RaStanza {
      "data": {
        "altColor": "120,12,12",
        "baseColorDefault": "genomicCodons",
        "baseColorUseCds": "given",
        "bigDataUrl": "bbi/GCF_000002985.6_WBcel235.ncbiRefSeq.bb",
        "color": "12,12,120",
        "defaultLabelFields": "geneName2",
        "idXref": "ncbiRefSeqLink mrnaAcc name",
        "labelFields": "name,geneName,geneName2",
        "longLabel": "NCBI RefSeq genes, curated and predicted sets (NM_*, XM_*, NR_*, XR_*, NP_* or YP_*)",
        "parent": "refSeqComposite on",
        "priority": "1",
        "searchIndex": "name",
        "searchTrix": "ixIxx/GCF_000002985.6_WBcel235.ncbiRefSeq.ix",
        "shortLabel": "RefSeq All",
        "track": "ncbiRefSeq",
        "type": "bigGenePred",
      },
      "name": "ncbiRefSeq",
      "nameKey": "track",
    },
    "ncbiRefSeqCurated": RaStanza {
      "data": {
        "baseColorDefault": "genomicCodons",
        "baseColorUseCds": "given",
        "bigDataUrl": "bbi/GCF_000002985.6_WBcel235.ncbiRefSeqCurated.bb",
        "color": "12,12,120",
        "defaultLabelFields": "geneName2",
        "idXref": "ncbiRefSeqLink mrnaAcc name",
        "labelFields": "name,geneName,geneName2",
        "longLabel": "NCBI RefSeq genes, curated subset (NM_*, NR_*, NP_* or YP_*)",
        "parent": "refSeqComposite on",
        "priority": "2",
        "searchIndex": "name",
        "searchTrix": "ixIxx/GCF_000002985.6_WBcel235.ncbiRefSeqCurated.ix",
        "shortLabel": "RefSeq Curated",
        "track": "ncbiRefSeqCurated",
        "type": "bigGenePred",
      },
      "name": "ncbiRefSeqCurated",
      "nameKey": "track",
    },
    "ncbiRefSeqOther": RaStanza {
      "data": {
        "bigDataUrl": "bbi/GCF_000002985.6_WBcel235.ncbiRefSeqOther.bb",
        "color": "32,32,32",
        "labelFields": "name",
        "longLabel": "NCBI RefSeq other annotations (not NM_*, NR_*, XM_*, XR_*, NP_* or YP_*)",
        "parent": "refSeqComposite on",
        "priority": "4",
        "searchIndex": "name",
        "searchTrix": "ixIxx/GCF_000002985.6_WBcel235.ncbiRefSeqOther.ix",
        "shortLabel": "RefSeq Other",
        "skipEmptyFields": "on",
        "track": "ncbiRefSeqOther",
        "type": "bigBed 12 +",
        "urls": "GeneID="https://www.ncbi.nlm.nih.gov/gene/$$" MIM="https://www.ncbi.nlm.nih.gov/omim/$$" HGNC="http://www.genenames.org/cgi-bin/gene_symbol_report?hgnc_id=$$" FlyBase="http://flybase.org/reports/$$" WormBase="http://www.wormbase.org/db/gene/gene?name=$$" RGD="https://rgd.mcw.edu/rgdweb/search/search.html?term=$$" SGD="https://www.yeastgenome.org/locus/$$" miRBase="http://www.mirbase.org/cgi-bin/mirna_entry.pl?acc=$$" ZFIN="https://zfin.org/$$" MGI="http://www.informatics.jax.org/marker/$$"",
      },
      "name": "ncbiRefSeqOther",
      "nameKey": "track",
    },
    "ncbiRefSeqPsl": RaStanza {
      "data": {
        "baseColorDefault": "diffCodons",
        "baseColorUseCds": "table given",
        "baseColorUseSequence": "lfExtra",
        "bigDataUrl": "bbi/GCF_000002985.6_WBcel235.bigPsl.bb",
        "color": "0,0,0",
        "indelDoubleInsert": "on",
        "indelQueryInsert": "on",
        "longLabel": "RefSeq Alignments of RNAs",
        "parent": "refSeqComposite off",
        "priority": "5",
        "pslSequence": "no",
        "searchIndex": "name",
        "shortLabel": "RefSeq Alignments",
        "showCdsAllScales": ".",
        "showCdsMaxZoom": "10000.0",
        "showDiffBasesAllScales": ".",
        "showDiffBasesMaxZoom": "10000.0",
        "track": "ncbiRefSeqPsl",
        "type": "bigPsl",
      },
      "name": "ncbiRefSeqPsl",
      "nameKey": "track",
    },
    "refSeqComposite": RaStanza {
      "data": {
        "allButtonPair": "on",
        "compositeTrack": "on",
        "dataVersion": "WormBase WS277 (2020-05-29)",
        "dragAndDrop": "subTracks",
        "group": "genes",
        "html": "html/GCF_000002985.6_WBcel235.refSeqComposite",
        "longLabel": "RefSeq gene predictions from NCBI",
        "priority": "2",
        "shortLabel": "NCBI RefSeq",
        "track": "refSeqComposite",
        "type": "bigBed",
        "visibility": "pack",
      },
      "name": "refSeqComposite",
      "nameKey": "track",
    },
    "repeatMasker": RaStanza {
      "data": {
        "bigDataUrl": "bbi/GCF_000002985.6_WBcel235.rmsk.bb",
        "group": "varRep",
        "html": "html/GCF_000002985.6_WBcel235.repeatMasker",
        "longLabel": "RepeatMasker Repetitive Elements",
        "maxWindowToDraw": "5000000",
        "shortLabel": "RepeatMasker",
        "track": "repeatMasker",
        "type": "bigRmsk 9 +",
        "visibility": "pack",
      },
      "name": "repeatMasker",
      "nameKey": "track",
    },
    "simpleRepeat": RaStanza {
      "data": {
        "bigDataUrl": "bbi/GCF_000002985.6_WBcel235.simpleRepeat.bb",
        "group": "varRep",
        "html": "html/GCF_000002985.6_WBcel235.simpleRepeat",
        "longLabel": "Simple Tandem Repeats by TRF",
        "shortLabel": "Simple Repeats",
        "track": "simpleRepeat",
        "type": "bigBed 4 +",
        "visibility": "dense",
      },
      "name": "simpleRepeat",
      "nameKey": "track",
    },
    "tanDups": RaStanza {
      "data": {
        "compositeTrack": "on",
        "group": "map",
        "html": "html/GCF_000002985.6_WBcel235.tanDups",
        "longLabel": "Paired identical sequences",
        "shortLabel": "Tandem Dups",
        "track": "tanDups",
        "type": "bigBed 12",
        "visibility": "hide",
      },
      "name": "tanDups",
      "nameKey": "track",
    },
    "tandemDups": RaStanza {
      "data": {
        "bigDataUrl": "bbi/GCF_000002985.6_WBcel235.tandemDups.bb",
        "longLabel": "Paired exactly identical sequence survey over entire genome assembly",
        "parent": "tanDups on",
        "shortLabel": "Tandem Dups",
        "track": "tandemDups",
        "type": "bigBed 12",
      },
      "name": "tandemDups",
      "nameKey": "track",
    },
    "windowMasker": RaStanza {
      "data": {
        "bigDataUrl": "bbi/GCF_000002985.6_WBcel235.windowMasker.bb",
        "group": "varRep",
        "html": "html/GCF_000002985.6_WBcel235.windowMasker",
        "longLabel": "Genomic Intervals Masked by WindowMasker + SDust",
        "shortLabel": "WM + SDust",
        "track": "windowMasker",
        "type": "bigBed 3",
        "visibility": "dense",
      },
      "name": "windowMasker",
      "nameKey": "track",
    },
    "xenoRefGene": RaStanza {
      "data": {
        "bigDataUrl": "bbi/GCF_000002985.6_WBcel235.xenoRefGene.bb",
        "color": "180,0,0",
        "defaultLabelFields": "geneName",
        "group": "rna",
        "html": "html/GCF_000002985.6_WBcel235.xenoRefGene",
        "labelFields": "name,geneName,geneName2",
        "longLabel": "RefSeq mRNAs mapped to this assembly",
        "searchIndex": "name",
        "searchTrix": "ixIxx/GCF_000002985.6_WBcel235.xenoRefGene.ix",
        "shortLabel": "RefSeq mRNAs",
        "track": "xenoRefGene",
        "type": "bigGenePred",
        "url": "https://www.ncbi.nlm.nih.gov/nuccore/$$",
        "urlLabel": "NCBI Nucleotide database:",
        "visibility": "pack",
      },
      "name": "xenoRefGene",
      "nameKey": "track",
    },
  },
  "nameKey": "track",
}
`;
